python/ray/data/_internal/stats.py

```diff
@@ -297,6 +308,17 @@ def clear_stats_actor_metrics(tags: Dict[str, str]):
     _stats_actor.clear_metrics.remote(tags)
 
 
+def get_dataset_id_from_stats_actor() -> str:
+    global _stats_actor
+    _check_cluster_stats_actor()
+    try:
+        return ray.get(_stats_actor.get_dataset_id.remote())
+    except ray.exceptions.RayError:
```

Comment:
Discussed offline, let's change this to catch all Python exception, so the dataset id creation never fails.